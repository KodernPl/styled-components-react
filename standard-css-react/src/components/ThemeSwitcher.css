@import '../styles/variables.css';

.theme-switcher {
  margin: 2rem;
  width: 4rem;
  height: 2rem;

  &__switch {
    cursor: pointer;
    text-indent: -999rem;
    width: 100%;
    height: 100%;
    display: block;
    border-radius: 10rem;
    position: relative;

    &::after {
      content: '';
      position: absolute;
      top: 10%;
      left: 5%;
      width: 50%;
      height: 80%;
      background: var(--color-white);
      border-radius: 50%;
      transition: 0.5s;
    }

    &--light {
      background: var(--color-sc-pink);
    }

    &--dark {
      background: var(--color-grey);

      &::after {
        left: calc(100% - .5rem);
        transform: translateX(-100%);
      }
    }
  }
}
