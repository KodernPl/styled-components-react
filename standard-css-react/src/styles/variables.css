:root {
  --color-white: white;
  --color-gradient1: #667db6;
  --color-gradient2: #0082c8;
  --color-gradient3: #0082c8;
  --color-gradient4: #667db6;
  --color-white: #fff;
  --color-black: #000;
  --color-dark-grey: #222;
  --color-darkest-grey: #111;
  --color-grey: grey;
  --color-palevioletred: palevioletred;
  --color-sc-pink: rgb(219, 112, 147);
  --color-sc-orange: rgb(218, 163, 87);
  --color-alpha-black: rgba(0, 0, 0, 0.2);
  --color-alpha-grey: rgba(20, 20, 20, 0.1);
  --gradient-light: linear-gradient(to top right, rgb(219, 112, 147), rgb(218, 163, 87));
  --gradient-dark: linear-gradient(to top right, #667db6, #0082c8, #0082c8);
}

@custom-media --desktop (min-width: 1024px);
