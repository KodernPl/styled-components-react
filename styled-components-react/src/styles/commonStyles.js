import styled from 'styled-components';

export const CenterElements = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export const CustomLink = styled.a`
  display: inline-block;
  width: 20rem;
  padding: .5rem 0;
  margin: .5rem 1rem;
  border-width: .2rem;
  border-radius: .3rem;
  border-style: solid;
  border-image: initial;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  font-size: 1.2rem;
`;
