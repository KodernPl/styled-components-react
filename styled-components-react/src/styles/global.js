import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  html {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: sans-serif;
  }

  body {
    margin: 0;
    padding: 0;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }
`;
