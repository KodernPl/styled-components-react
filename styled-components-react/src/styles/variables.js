export const COLORS = {
  gradient1: '#667db6',
  gradient2: '#0082c8',
  gradient3: '#0082c8',
  gradient4: '#667db6',

  white: '#fff',
  black: '#000',
  darkGrey: '#222',
  darkestGrey: '#111',

  grey: 'grey',
  palevioletred: 'palevioletred',

  scPink: 'rgb(219, 112, 147)',
  scOrange: 'rgb(218, 163, 87)',

  alphaBlack: 'rgba(0, 0, 0, 0.2)',
  alphaGrey: 'rgba(20, 20, 20, 0.1)',
};

export const THEMES = {
  light: 'light',
  dark: 'dark',
};

export const GRADIENTS = {
  light: `linear-gradient(to top right, ${COLORS.scPink}, ${COLORS.scOrange})`,
  dark: `linear-gradient(to top right, ${COLORS.gradient1}, ${COLORS.gradient2}, ${COLORS.gradient4})`,
};

export const BREAKPOINTS = {
  desktop: '(min-width: 1024px)',
};
