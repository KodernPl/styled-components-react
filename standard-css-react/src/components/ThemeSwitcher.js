import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { THEMES } from '../styles/variables';

import './ThemeSwitcher.css';

const ns = 'theme-switcher';
const displayName = 'ThemeSwitcher';

const propTypes = {
  onThemeChange: PropTypes.func.isRequired,
  theme: PropTypes.string.isRequired,
};

const ThemeSwitcher = ({ theme, onThemeChange }) => {
  const switchClassNames = cx(
    [`${ns}__switch`],
    {
      [`${ns}__switch--dark`]: theme === THEMES.dark,
      [`${ns}__switch--light`]: theme === THEMES.light,
    },
  );
  return (
    <div className={ns}>
      <div className={switchClassNames} onClick={onThemeChange} />
    </div>
  );
};

ThemeSwitcher.displayName = displayName;
ThemeSwitcher.propTypes = propTypes;

export default ThemeSwitcher;
