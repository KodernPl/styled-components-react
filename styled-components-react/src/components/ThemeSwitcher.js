import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { COLORS, THEMES } from '../styles/variables';

const displayName = 'ThemeSwitcher';

const ThemeSwitcherWrapper = styled.div`
  margin: 2rem;
  width: 4rem;
  height: 2rem;
`;

const Switch = styled.div`
  cursor: pointer;
  text-indent: -999rem;
  width: 100%;
  height: 100%;
  background: ${COLORS.scPink};
  display: block;
  border-radius: 10rem;
  position: relative;

  &::after {
    content: '';
    position: absolute;
    top: 10%;
    left: 5%;
    width: 50%;
    height: 80%;
    background: ${COLORS.white};
    border-radius: 50%;
    transition: 0.5s;
  }

  ${({ checked }) => (checked && css`
      &::after {
        left: calc(100% - .5rem);
        transform: translateX(-100%);
      }
      background: ${COLORS.grey};
  `)}
`;

const propTypes = {
  onThemeChange: PropTypes.func.isRequired,
  theme: PropTypes.string.isRequired,
};

const ThemeSwitcher = ({ theme, onThemeChange }) => (
  <ThemeSwitcherWrapper>
    <Switch onClick={onThemeChange} checked={theme === THEMES.dark} />
  </ThemeSwitcherWrapper>
);

ThemeSwitcher.displayName = displayName;
ThemeSwitcher.propTypes = propTypes;

export default ThemeSwitcher;
