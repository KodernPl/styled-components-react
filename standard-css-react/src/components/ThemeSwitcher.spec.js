import React from 'react';
import { mount } from 'enzyme';
import { THEMES } from '../styles/variables';
import ThemeSwitcher from './ThemeSwitcher';

const mockOnThemeChangeFn = jest.fn();

const renderComponent = ({
  theme = THEMES.light,
}) => mount(
  <ThemeSwitcher
    theme={theme}
    onThemeChange={mockOnThemeChangeFn}
  />,
);

beforeEach(() => {
  mockOnThemeChangeFn.mockReset();
});

[
  THEMES.light,
  THEMES.dark,
].forEach((theme) => {
  describe(`${theme} theme`, () => {
    test(`renders correct HTML for ${theme} theme`, () => {
      expect(renderComponent({ theme })).toMatchSnapshot();
    });
  });
});
