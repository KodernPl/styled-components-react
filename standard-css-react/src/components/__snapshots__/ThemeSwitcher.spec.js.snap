// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark theme renders correct HTML for dark theme 1`] = `
<ThemeSwitcher
  onThemeChange={[MockFunction]}
  theme="dark"
>
  <div
    className="theme-switcher"
  >
    <div
      className="theme-switcher__switch theme-switcher__switch--dark"
      onClick={[MockFunction]}
    />
  </div>
</ThemeSwitcher>
`;

exports[`light theme renders correct HTML for light theme 1`] = `
<ThemeSwitcher
  onThemeChange={[MockFunction]}
  theme="light"
>
  <div
    className="theme-switcher"
  >
    <div
      className="theme-switcher__switch theme-switcher__switch--light"
      onClick={[MockFunction]}
    />
  </div>
</ThemeSwitcher>
`;
